File: Arrays/Easy/283/283.py
from typing import List

class Solution:
    def moveZeroes(self, nums: List[int]) -> None:
        """
        Do not return anything, modify nums in-place instead.
        """
        n = len(nums)
        slow = 0
        fast = 0
        while fast < n:
            if nums[fast] != 0:
                nums[slow], nums[fast] = nums[fast], nums[slow]
                slow += 1
            fast += 1
        while slow < n:
            nums[slow] = 0
            slow += 1

File: Arrays/Easy/283/283.c
void moveZeroes(int* nums, int numsSize){
    int slow = 0;
    int fast = 0;
    while(fast < numsSize){
        if(nums[fast] != 0){
            int temp = nums[fast];
            nums[fast] = nums[slow];
            nums[slow] = temp;
            slow++;
        }
        fast++;
    }
    while(slow < numsSize){
        nums[slow] = 0;
        slow++;
    }
}